src/gettext.h

```diff
@@ -77,3 +78,20 @@ inline std::wstring fwgettext(const char *src, Args&&... args)
 	delete[] str;
 	return std::wstring(buf);
 }
+
+template <typename ...Args>
+inline std::string fmtgettext(const char *format, std::size_t buf_size, Args&&... args)
```

Comment:
Also if it can be automatically detected why have `buf_size` at all? You could just remove the parameter.

Reply:
For performance purpose. It will execute `mt_snprintf` once If you know the max bufer size.

Reply:
I thought about this more and am still against it.
Even if the source string is really short (and parameters bounded) you cannot predict how big the string will when translated. This is especially relevant since e.g. each Chinese character is 3 bytes under UTF-8.

I have an idea for your concern though:
* Allocate a 128 byte `std::string` in the beginning
* Call snprintf and check if the string is cut off
* If the buffer was large enough, return it
* Otherwise double the size and go to step 2

Reply:
Enh, I have understood what you mean.